src/pybind/mgr/cephadm/serve.py

```diff
@@ -501,7 +501,8 @@ def _check_for_strays(self) -> None:
                                     s.get('type'), s.get('id')
                                 )
                             )
-
+                    if s.get('type') == 'tcmu-runner':
```

Comment:
Although this does resolve the stray daemon warning that was happening. This kinda feels weird. Essentially this compares info from list_servers() to the same info from list_servers() so it will never fail the check on line 509. 
I'm not sure where else we would get info about the tcmu-runner containers to make a better comparison? Maybe it would be better to just ignore tcmu-runner containers?

Reply:
Ah could you explain what you mean by it's comparing info from list_servers() to same info from list_servers()? I thought the issue though was that although `tcmu-runner` is in list_servers(), when I look at `managed` which is from `self.mgr.cache.get_daemon_names()`, I don't see `tcmu-runner`, but it should be considered managed by cephadm. 

```
Nov 06 02:19:37 vm-00 ceph-mgr[18221]: log_channel(cephadm) log [INF] : ls
                                       [{'hostname': 'vm-00', 'services': [{'type': 'mgr', 'id': 'vm-00.tjjcta'}, {'type': 'mon', 'id': 'vm-00'}, {'type': 'osd', 'id': '0'}, {'type': 'tcmu-runner', 'id': 'vm-00:rbd/disk_1'}], 'ceph_version': 'ceph version 17.0.0-8135-gf5b96461 (f5b96461081c2f508d1393202ebcd94e6bd2ea3f) quincy (dev)'}, {'hostname': 'vm-01', 'services': [{'type': 'mgr', 'id': 'vm-01.nzxjid'}, {'type': 'mon', 'id': 'vm-01'}, {'type': 'osd', 'id': '1'}], 'ceph_version': 'ceph version 17.0.0-8135-gf5b96461 (f5b96461081c2f508d1393202ebcd94e6bd2ea3f) quincy (dev)'}, {'hostname': 'vm-02', 'services': [{'type': 'mon', 'id': 'vm-02'}, {'type': 'osd', 'id': '2'}], 'ceph_version': 'ceph version 17.0.0-8135-gf5b96461 (f5b96461081c2f508d1393202ebcd94e6bd2ea3f) quincy (dev)'}]
Nov 06 02:19:37 vm-00 ceph-mgr[18221]: [cephadm INFO cephadm.serve] managed
                                       ['mon.vm-00', 'mgr.vm-00.tjjcta', 'alertmanager.vm-00', 'crash.vm-00', 'grafana.vm-00', 'node-exporter.vm-00', 'osd.0', 'prometheus.vm-00', 'iscsi.iscsi.vm-00.ynpymj', 'crash.vm-01', 'mgr.vm-01.nzxjid', 'mon.vm-01', 'node-exporter.vm-01', 'osd.1', 'iscsi.iscsi.vm-01.svhijq', 'crash.vm-02', 'mon.vm-02', 'node-exporter.vm-02', 'osd.2', 'iscsi.iscsi.vm-02.snrszo']
```

By ignoring tcmu-runner containers do you mean doing something like `if name not in managed and not name.startswith('tcmu-runner'.)`?

Reply:
@melissa-kun-li Ultimately, this "managed" list is just a temporary list we're using for the purpose of finding stray daemons (before this change it's just equivalent to the daemons in the cache but whatever) so checking if the daemon is in managed or is type tcmu-runner versus just adding all the tcmu-runner daemons to managed are equivalent.  I feel like the point Dan was making is it would be nice to check somehow if the tcmu-runner daemon we find is actually part of one of the iscsi services we managed (or some other check to try to actually verify we actually manage this daemon rather than blanket accepting tcmu-runner daemons as managed). The problem with this is, from the tracker, all we're going to get is tcmu-runner daemons with names like `tcmu-runner.host2.domain.com:iscsi-pool-0001/iscsi-p0001-img-01` and, like Dan,  I don't see what info we could pull from this to try and tell if verify if we "manage" this particular tcmu-runner daemon (other than maybe checking the hostname in the daemon name, but that's already done here via the stray host check). We don't track tcmu-runner daemons in the host cache so there's nothing to compare to. I really can't think of any good check to put in for these tcmu-runner daemons unless we refactor the way we handle the iscsi service so I think blanket accepting them as managed is okay

Tldr, I think this PR is good as soon as the test is added.

Reply:
Ohh I see what you and Daniel are saying, thanks for explaining Adam! I didn't consider that there may be tcmu-runner daemons outside of the iscsi service we manage. Ah and I think I see what Daniel means by ignoring the tcmu-runner containers. Hmm I'll think if there's a better solution, in the meantime I'll work on the test

Reply:
ping?